### PR TITLE
Illegal tech rework

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Illegal tech is now boosted by an illegal item flag which is provided to items upon being purchased from the traitor uplink.

## Why It's Good For The Game

Prevents mapped items that happen to be in the uplink from allowing illegal tech to be cheesed.

## Changelog
:cl:
refactor: Illegal tech is now given to items bought from the uplink, rather than items that exist in the uplink.
/:cl:

### CODE REVIEW

**Review by ike709 — APPROVED** [review]

**Comment by KittyNoodle** [comment]
no more roundstart illegals :(

**Comment by Cenrus** [comment]
You can do away with the `illegal_tech` var from uplink item datums with this change.

**Comment by PowerfulBacon** [comment]
> 
> 
> You can do away with the `illegal_tech` var from uplink item datums with this change.

Oh true

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 1]
```diff
@@ -380,6 +380,7 @@
 	allowed = list(/obj/item/gun, /obj/item/ammo_box,/obj/item/ammo_casing, /obj/item/melee/baton, /obj/item/melee/transforming/energy/sword/saber, /obj/item/restraints/handcuffs, /obj/item/tank/internals)
 	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/syndi
 	jetpack = /obj/item/tank/jetpack/suit
+	item_flags = ILLEGAL	//Syndicate only and difficult to obtain outside of uplink anyway. Nukie hardsuits on the ship are illegal.
```
Comment (Archanial) [inline code/modules/clothing/spacesuits/hardsuit.dm:383]: It's rare loot on lavaland

File: code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -153,11 +153,14 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
-			to_chat(H, "[A] materializes into your hands!")
-			return A
+	if(istype(A, /obj/item))
+		var/obj/item/I = A
+		I.item_flags |= ILLEGAL
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			if(H.put_in_hands(A))
+				to_chat(H, "[A] materializes into your hands!")
+				return A
```
Comment (Archanial) [inline code/modules/uplink/uplink_items.dm:163]:
```suggestion
	if(ishuman(user))
		var/mob/living/carbon/human/H = user
		if(H.put_in_hands(A))
			to_chat(H, "[A] materializes into your hands!")
			return A
```
those two checks are unrelated

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 1 above)
Comment (BrumalLake) [reply code/modules/clothing/spacesuits/hardsuit.dm:383]: miner aint gonna sacrifice their super suit for some illegal tech on the station
its called  being selfish

File: code/modules/uplink/uplink_items.dm (hunk 2 above)
Comment (PowerfulBacon) [reply code/modules/uplink/uplink_items.dm:163]: No they are not, see the removed diff

**Comment by wraith-54321** [comment]
aww, no more floor contraband
